t/semaphore.t

```diff
@@ -0,0 +1,1431 @@
+                end
+                ngx.semaphore.test = sem
+            end
+            local sem  = ngx.semaphore.test
+            local ok,err = sem:wait(10)
+            if ok then
+                ngx.print("wait")
+                ngx.exit(200)
+            else
+                ngx.exit(500)
+            end
+        ';
+    }
+
+    location /sem_post {
+        content_by_lua '
+            require "resty.core.semaphore"
+            if not ngx.semaphore.test then
+                local sem,err = ngx.semaphore.new(0)
+                if not sem then
+                    ngx.say(err)
+                    ngx.exit(500)
+                end
+                ngx.semaphore.test = sem
+            end
+            local sem  = ngx.semaphore.test
+            local ok,err = sem:post()
+            if ok then
+                ngx.print("post")
+                ngx.exit(200)
+            else
+                ngx.exit(500)
+            end
+
+    ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+200
+wait
+200
+post
+--- no_error_log
+[error]
+
+
+
+=== TEST 2: basic semaphore wait post
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        content_by_lua '
+            require "resty.core.semaphore"
+            local sem  = ngx.semaphore.new(0)
+            local ok,err=sem:wait(1)
+            ngx.say(tostring(ok).." "..tostring(err))
+        ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+false timeout
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: basic semaphore wait not allow in body_filter_by_lua
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        body_filter_by_lua '
+            require "resty.core.semaphore"
+            local sem = ngx.semaphore.new(0)
+            local ok,err=sem:wait(1)
+            ngx.log(ngx.ERR,err)
+        ';
+        return 200;
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+--- error_log
+API disabled in the context of (unknown)
+
+
+
+=== TEST 4: basic semaphore wait not allow in header_filter_by_lua
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        header_filter_by_lua '
+            require "resty.core.semaphore"
+            local sem  = ngx.semaphore.new(0)
+            local ok,err=sem:wait(1)
+            ngx.log(ngx.ERR,err)
+        ';
+        return 200;
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+
+
+
+--- error_log
+API disabled in the context of header_filter_by_lua*
+
+=== TEST 5: basic semaphore wait not allow in init_worker_by_lua
+--- http_config eval: $::HttpConfigIntWorkerBy
+--- config
+    location /test {
+       content_by_lua '
+            require "resty.core.semaphore"
+            ngx.say(ngx.semaphore.test)
+       ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+API disabled in the context of init_worker_by_lua*
+--- no_error_log
+[error]
+
```

Comment:
Style: please use the [reindex](https://raw.githubusercontent.com/agentzh/old-openresty/master/bin/reindex) script to reformat your `.t` files so that it's consistently formatted. Thanks.